spring-cloud-gateway-integration-tests/grpc/src/main/java/org/springframework/cloud/gateway/tests/grpc/GRPCApplication.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2013-2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.gateway.tests.grpc;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.concurrent.TimeUnit;
+
+import io.grpc.Grpc;
+import io.grpc.Server;
+import io.grpc.ServerCredentials;
+import io.grpc.TlsServerCredentials;
+import io.grpc.stub.StreamObserver;
+
+import org.springframework.boot.ApplicationArguments;
+import org.springframework.boot.ApplicationRunner;
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.SpringBootConfiguration;
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
+import org.springframework.cloud.gateway.route.RouteLocator;
+import org.springframework.cloud.gateway.route.builder.RouteLocatorBuilder;
+import org.springframework.context.annotation.Bean;
+import org.springframework.core.io.ClassPathResource;
+import org.springframework.stereotype.Component;
+import org.springframework.util.SocketUtils;
+
+/**
+ * @author Alberto C. Ríos
+ */
+@SpringBootConfiguration
+@EnableAutoConfiguration
+public class GRPCApplication {
+
+	private static final int GRPC_SERVER_PORT = SocketUtils.findAvailableTcpPort();
```

Comment:
circleci is not happy about opening a channel on the found ports. Is there anything I am missing?


```
Starting server in port 20735 2021-10-06 12:25:31.349 WARN 9056 --- [or-http-epoll-2] io.netty.channel.ChannelInitializer : Failed to initialize a channel. Closing: [id: 0xf8106f12, L:/127.0.0.1:38150 - R:/127.0.0.1:50622] java.lang.RuntimeException: ALPN unsupported. Is your classpath configured correctly? For Conscrypt, add the appropriate Conscrypt JAR to classpath and set the security provider. For Jetty-ALPN, see https://www.eclipse.org/jetty/documentation/current/alpn-chapter.html#alpn-starting at io.netty.handler.ssl.JdkAlpnApplicationProtocolNegotiator$FailureWrapper.wrapSslEngine(JdkAlpnApplicationProtocolNegotiator.java:122) ~[netty-handler-4.1.67.Final.jar:4.1.67.Final] at io.netty.handler.ssl.JdkSslContext.configureAndWrapEngine(JdkSslContext.java:351) ~[netty-handler-4.1.67.Final.jar:4.1.67.Final] at io.netty.handler.ssl.JdkSslContext.newEngine(JdkSslContext.java:321) ~[netty-handler-4.1.67.Final.jar:4.1.67.Final] at io.netty.handler.ssl.SslContext.newHandler(SslContext.java:960) ~[netty-handler-4.1.67.Final.jar:4.1.67.Final] at io.netty.handler.ssl.SslContext.newHandler(SslContext.java:952) ~[netty-handler-4.1.67.Final.jar:4.1.67.Final] at reactor.netty.tcp.SslProvider.addSslHandler(SslProvider.java:542) ~[reactor-netty-core-1.0.10.jar:1.0.10] at reactor.netty.http.server.HttpServerConfig$HttpServerChannelInitializer.onChannelInit(HttpServerConfig.java:874) ~[reactor-netty-http-1.0.10.jar:1.0.10] at reactor.netty.transport.TransportConfig$TransportChannelInitializer.initChannel(TransportConfig.java:394) ~[reactor-netty-core-1.0.10.jar:1.0.10] at io.netty.channel.ChannelInitializer.initChannel(ChannelInitializer.java:129) [netty-transport-4.1.67.Final.jar:4.1.67.Final] at io.netty.channel.ChannelInitializer.handlerAdded(ChannelInitializer.java:112) [netty-transport-4.1.67.Final.jar:4.1.67.Final] at io.netty.channel.AbstractChannelHandlerContext.callHandlerAdded(AbstractChannelHandlerContext.java:938) [netty-transport-4.1.67.Final.jar:4.1.67.Final] at io.netty.channel.DefaultChannelPipeline.callHandlerAdded0(DefaultChannelPipeline.java:609) [netty-transport-4.1.67.Final.jar:4.1.67.Final] at io.netty.channel.DefaultChannelPipeline.access$100(DefaultChannelPipeline.java:46) [netty-transport-4.1.67.Final.jar:4.1.67.Final] at io.netty.channel.DefaultChannelPipeline$PendingHandlerAddedTask.execute(DefaultChannelPipeline.java:1463) [netty-transport-4.1.67.Final.jar:4.1.67.Final] at io.netty.channel.DefaultChannelPipeline.callHandlerAddedForAllHandlers(DefaultChannelPipeline.java:1115) [netty-transport-4.1.67.Final.jar:4.1.67.Final] at io.netty.channel.DefaultChannelPipeline.invokeHandlerAddedIfNeeded(DefaultChannelPipeline.java:650) [netty-transport-4.1.67.Final.jar:4.1.67.Final] at io.netty.channel.AbstractChannel$AbstractUnsafe.register0(AbstractChannel.java:514) [netty-transport-4.1.67.Final.jar:4.1.67.Final] at io.netty.channel.AbstractChannel$AbstractUnsafe.access$200(AbstractChannel.java:429) [netty-transport-4.1.67.Final.jar:4.1.67.Final] at io.netty.channel.AbstractChannel$AbstractUnsafe$1.run(AbstractChannel.java:486) [netty-transport-4.1.67.Final.jar:4.1.67.Final] at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:164) [netty-common-4.1.67.Final.jar:4.1.67.Final] at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:469) [netty-common-4.1.67.Final.jar:4.1.67.Final] at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:387) [netty-transport-native-epoll-4.1.67.Final-linux-x86_64.jar:4.1.67.Final] at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:986) [netty-common-4.1.67.Final.jar:4.1.67.Final] at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) 
```